### PR TITLE
WIP Optimize j9shr_storeGCHints() to avoid storing unchanged data

### PR DESCRIPTION
**Pull request by pshipton**
Related to #4168 and #3743

Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity zlinux jdk11

**Comment by pshipton** [comment]
The zlinux testing passed. Rebasing to pick up the Windows compile fix https://github.com/eclipse/openj9/pull/4213 so Windows can be tested.

**Comment by pshipton** [comment]
jenkins test sanity win jdk11

File: runtime/shared_common/shrinit.cpp [hunk 1]
```diff
@@ -5082,12 +5082,13 @@ void
...
 	J9JavaVM* vm = currentThread->javaVM;
-	bool heapSizesSet = J9_ARE_ALL_BITS_SET(vm->sharedClassConfig->localStartupHints.hintsData.flags, J9SHR_STARTUPHINTS_HEAPSIZES_SET);
+	J9SharedStartupHintsDataDescriptor* hintsData = &vm->sharedClassConfig->localStartupHints.hintsData;
+	bool heapSizesSet = J9_ARE_ALL_BITS_SET(hintsData->flags, J9SHR_STARTUPHINTS_HEAPSIZES_SET);
 
-	if (forceReplace || !heapSizesSet) {
-		vm->sharedClassConfig->localStartupHints.hintsData.heapSize1 = heapSize1;
-		vm->sharedClassConfig->localStartupHints.hintsData.heapSize2 = heapSize2;
-		vm->sharedClassConfig->localStartupHints.hintsData.flags |= J9SHR_STARTUPHINTS_HEAPSIZES_SET;
+	if (!heapSizesSet || (forceReplace && ((heapSize1 != hintsData->heapSize1) || (heapSize2 != hintsData->heapSize2)))) {
```
Comment (keithc-ca) [inline runtime/shared_common/shrinit.cpp]: This changes the meaning of the parameter `forceReplace`: the name should be changed to match the new behavior (here and where this function and pointers to it are declared).

**Pull request closed without merging**